pkg/snapstore/s3_snapstore.go

```diff
@@ -495,30 +495,27 @@ func (s *S3SnapStore) Delete(snap brtypes.Snapshot) error {
 	return err
 }
 
-// S3SnapStoreHash calculates and returns the hash of aws S3 snapstore secret.
-func S3SnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
-	if dir, isSet := os.LookupEnv(awsCredentialFile); isSet {
-		awsConfig, err := readAWSCredentialFromDir(dir)
+// GetS3CredentialsLastModifiedTime returns the latest modification timestamp of the AWS credential file(s)
+func GetS3CredentialsLastModifiedTime() (time.Time, error) {
+	if dir, isSet := os.LookupEnv(awsCredentialDirectory); isSet {
+		// credential files which are essential for creating the S3 snapstore
+		credentialFiles := []string{"accessKeyID", "region", "secretAccessKey"}
+		awsTimeStamp, err := getLatestFileModificationTime(dir, credentialFiles)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("failed to get AWS credential timestamp from the directory %v with error: %v", dir, err)
 		}
-		return getS3Hash(awsConfig), nil
+		return awsTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(awsCredentialJSONFile); isSet {
-		awsConfig, err := credentialsFromJSON(filename)
+		fileInfo, err := os.Stat(filename)
```

Comment:
You could have just created a single function which could be used everywhere.
```go
func getLatestCredentialsModifiedTime(paths ...string) (time.Time, error) {
}
```

Reply:
Will be following this to generalize checking file timestamp check across both mounted files in a directory and JSON.